app/main/user-agent-client.js

```diff
@@ -0,0 +1,87 @@
+/*
+ Copyright 2016 Mozilla
+
+ Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ this file except in compliance with the License. You may obtain a copy of the
+ License at http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software distributed
+ under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ specific language governing permissions and limitations under the License.
+ */
+
+import path from 'path';
+import EventEmitter from 'events';
+import WebSocket from 'ws';
+import { setTimeout } from 'timers';
+import childProcess from 'child_process';
+import * as endpoints from '../shared/constants/endpoints';
+
+class UserAgent extends EventEmitter {
+  constructor(ws, childCount) {
+    this.ws = ws;
+    this.childCount = childCount;
+
+    this.on('message', (type, args) => {
+      this.emit(type, ...args);
+    });
+  }
+}
+
+const timeout = (ms) => new Promise(resolve => setTimeout(resolve, ms));
+
+function startService() {
+  // Runs Electron as a node process
```

Comment:
Well, the one thing we know a tofino user has installed is the current electron build. They may not have node or anything else unless we add that as a dependency and ship it.